src/iadmin.cpp

```diff
@@ -147,6 +149,140 @@ auto get_replica_value(
     return v;
 } // get_replica_value
 
+auto check_that_connected_user_is_rodsadmin(RcComm* _comm) -> bool
+{
+    if (!_comm) {
+        return false;
+    }
+
+    try {
+        namespace adm = irods::experimental::administration;
+
+        const auto user_type =
+            adm::client::type(*_comm, adm::user{_comm->clientUser.userName, _comm->clientUser.rodsZone});
+
+        if (!user_type) {
+            rodsLogError(
+                LOG_ERROR, CAT_INVALID_USER_TYPE, "Could not determine if user has permission to perform operation.");
+            return false;
+        }
+
+        if (*user_type != adm::user_type::rodsadmin) {
+            rodsLogError(LOG_ERROR, CAT_INSUFFICIENT_PRIVILEGE_LEVEL, "Operation requires rodsadmin level privileges.");
+            return false;
+        }
+    }
+    catch (const irods::exception& e) {
+        rodsLogError(LOG_ERROR, e.code(), e.client_display_what());
+        return false;
+    }
+    catch (const std::exception& e) {
+        rodsLogError(LOG_ERROR, SYS_UNKNOWN_ERROR, e.what());
+        return false;
+    }
+
+    return true;
+} // check_that_connected_user_is_rodsadmin
+
+auto print_pam_password_config() -> int
+{
+    if (!check_that_connected_user_is_rodsadmin(Conn)) {
```

Comment:
Consider renaming so that usage reads like a question.

- `is_connected_as_rodsadmin`
- `is_rodadmin`
- `is_privileged_user`
- `connected_user_is_rodsadmin`

Those are just some examples. Feel free to come up with something else.

Reply:
The idea behind the name was to hint at side effects (the function will print messages for various errors that occur) and not give the impression that it's a pure function. I see that it did not do its job well :p 

If we're okay with the side effects, I'll just rename it to `connected_user_is_rodsadmin` which is what I had originally. Otherwise, I'll think about a better approach here (to me, top candidate is just not using a function).

Reply:
The side effect of the function which is only supposed to return whether the connected user is a rodsadmin is that it also prints messages for various errors that occur.

At least in my mind, a "clean" implementation of this function would be to only return whether the connected user is a rodsadmin and leave it up to the caller to print errors and whatnot. That's all I was getting at there. Might not be a big deal that the function prints things - I mostly did this because I didn't want to see all that mess in `print_pam_password_config`

Reply:
I've decided to just remove this function. Some code duplication now, but at least there's no hidden surprises :)

Reply:
This function has been removed. Resolving.